src/status_im/contexts/wallet/add_account/create_account/view.cljs

```diff
@@ -57,125 +54,216 @@
+           :on-change-text on-change-text}]
+         [color-picker
+          {:account-color     @account-color
+           :set-account-color set-account-color}]
+         [new-account-origin
+          {:derivation-path     constants/wallet-initial-derivation-path
+           :customization-color customization-color
+           :keypair-title       keypair-name}]]))))
+
+(defn derive-account-variant
+  [{:keys [on-change-text set-account-color set-emoji]
+    {:keys [account-name account-color emoji]}
+    :state}]
+  (let [derivation-path     (reagent/atom "")
+        set-derivation-path #(reset! derivation-path %)]
+    (fn [{:keys [customization-color]}]
+      (let [{:keys [derived-from
+                    key-uid]} (rf/sub [:wallet/selected-keypair])
+            on-auth-success   (rn/use-callback
+                               (fn [password]
+                                 (let [preferences {:account-name @account-name
+                                                    :color        @account-color
+                                                    :emoji        @emoji}]
+                                   (rf/dispatch
+                                    [:wallet/derive-address-and-add-account
+                                     {:password             password
+                                      :derived-from-address derived-from
+                                      :derivation-path      @derivation-path
+                                      :account-preferences  preferences}])))
+                               [derived-from])]
+        (rn/use-effect
+         #(rf/dispatch
+           [:wallet/next-derivation-path
+            {:on-success  set-derivation-path
+             :keypair-uid key-uid}])
+         [key-uid])
+
+        [floating-button
+         {:account-color      @account-color
+          :slide-button-props {:on-auth-success on-auth-success
+                               :disabled?       (or (empty? @account-name)
+                                                    (= "" @derivation-path))}}
+         [avatar
+          {:account-color   @account-color
+           :emoji           @emoji
+           :on-select-emoji set-emoji}]
+         [input
+          {:account-color  @account-color
+           :account-name   @account-name
+           :on-change-text on-change-text}]
+         [color-picker
+          {:account-color     @account-color
+           :set-account-color set-account-color}]
+         [new-account-origin
+          {:derivation-path     @derivation-path
+           :customization-color customization-color}]]))))
+
+(defn view
+  [_]
+  (let [account-name      (reagent/atom "")
+        account-color     (reagent/atom (rand-nth colors/account-colors))
+        emoji             (reagent/atom (emoji-picker.utils/random-emoji))
+        on-change-text    #(reset! account-name %)
+        set-account-color #(reset! account-color %)
+        set-emoji         #(reset! emoji %)
```

Comment:
Since we are at it, these reagent atoms can be refactored to react's state

Reply:
Well, I mostly refactored the state management instead of re-think it, I also thought about it but they aren't equivalent.

As you can see, I'm passing atoms to the children components, and I'm just `deref`ing inside the children, so that this component is not re-rendered when ratoms change.

If we change this approach to react's use-state, we will re-render everything on each value changed, we could try to get a similar result, but that's a bit out of scope of this PR.

There are some other screens still using ratoms, IMO this screen can be changed when we focus on completely remove ratoms from the codebase.